client/client.go

```diff
@@ -38,20 +41,21 @@ func lookupSrv(name string) ([]string, error) {
 	}
 	_, srvs, err := net.LookupSRV("", "", name)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("%w: %s", errNotAnSRV, err)
 	}
 	var addrs []string
 	for _, srv := range srvs {
 		hosts, err := net.LookupHost(srv.Target)
-		if err != nil {
-			return nil, err
-		}
-		if len(hosts) == 0 {
-			return nil, fmt.Errorf("unknown error looking up host %v for srv record %v", srv.Target, name)
+		// skip hosts that can't be resolved
+		if err != nil || len(hosts) == 0 {
+			continue
 		}
 		host := net.JoinHostPort(hosts[0], strconv.Itoa(int(srv.Port)))
```

Comment:
Why do we only append one of the addresses of the host?
Why not put the `srv.Target` into the list? The Dial function could then resolve it again and use any of the addresses.

Reply:
I don't think I got your question. Would mind elaborating? How is that one related to the PR?

Reply:
I see. I think the logic makes sense: an SRV record could contain a slice of targets, and sometimes the host could be broken and unresolvable. Because we have a slice of hosts, we can filter out those broken ones and continue issuing requests correctly, instead of failing during dialling to the host. Also, we do need to obey the srv.Port that is advertized by an SRV.